src/neoxp/Extensions/Extensions.cs

```diff
@@ -266,5 +267,48 @@ static bool TryGetNEP6Wallet(string path, string password, ProtocolSettings sett
                 }
             }
         }
+        
+        
+        private static JObject[] MakeFindStatesParams(
+            UInt256 rootHash,
+            UInt160 scriptHash,
+            ReadOnlySpan<byte> prefix,
+            ReadOnlySpan<byte> from = default (ReadOnlySpan<byte>),
+            int? count = null)
+        {
+            JObject[] jobjectArray = new JObject[count.HasValue ? 5 : 4];
+            jobjectArray[0] = (JObject) rootHash.ToString();
+            jobjectArray[1] = (JObject) scriptHash.ToString();
+            jobjectArray[2] = (JObject) Convert.ToBase64String(prefix);
+            jobjectArray[3] = (JObject) Convert.ToBase64String(from);
+            if (count.HasValue)
+                jobjectArray[4] = (JObject) (double) count.Value;
+            return jobjectArray;
+        }
+        public static async Task<ExpressRpcFoundStates> ExpressFindStatesAsync(this RpcClient rpcClient, UInt256 rootHash, UInt160 contractScriptHash, ReadOnlyMemory<byte> prefix, ReadOnlyMemory<byte> from = default, int? pageSize = null)
```

Comment:
We only use the `Results` field from `ExpressRpcFoundStates` so lets just return that. Also, if we return a `Task<IEnumerable<(string key, string value)>>` we can avoid interim `ToArray` calls - ExpressFindStatesAsync can simply concat the results of each `findstates` call together an the caller can call `ToArray` on the return value

Reply:
I like the idea, but I can't get this to work. Something about the function body that can't be an iterator block due to the return type not being an iterator interface type (probably due to `Task<>`). Can you do this change instead?

Reply:
Will do